cpp/core/utils/TaskContext.cc

```diff
@@ -19,20 +19,49 @@
 #include <folly/Likely.h>
 #include "utils/exception.h"
 
-namespace gluten {
+#include <iostream>
+#include <mutex>
+
+namespace {
+class TaskContextStorage {
+ public:
+  virtual ~TaskContextStorage() {
+    for (auto itr = objects_.rbegin(); itr != objects_.rend(); itr++) {
+      itr->reset();
+    }
+  }
+
+  void bind(std::shared_ptr<void> object) {
+    objects_.push_back(object);
+  }
+
+ private:
+  std::vector<std::shared_ptr<void>> objects_;
+};
 
 thread_local std::unique_ptr<TaskContextStorage> taskContextStorage = nullptr;
+static std::unique_ptr<TaskContextStorage> fallbackStorage = std::make_unique<TaskContextStorage>();
+static std::mutex fallbackStorageMutex;
+} // namespace
 
-void bindToTask(std::shared_ptr<void> object) {
-  GLUTEN_CHECK(taskContextStorage != nullptr, "Not in a Spark task");
-  taskContextStorage->bind(object);
+namespace gluten {
+
+bool onSparkTaskMainThread() {
```

Comment:
use `isXyz()` to represent checking meaning?

Reply:
OK.